### PR TITLE
Visual editor panel (cosmetic) refactoring

### PR DESCRIPTION
**Pull request by xulman**
This is supposed to be the solution to the issue #23  (at least the first commit is)

then I noticed public getter returing directly some private attribute and thought such getter is unnecessary (does not secure/protect anything), besides the getter was ill-named and should be deprecated anyway; I hope you'd like it ;-)

### CODE REVIEW

File: src/main/java/org/scijava/ui/behaviour/io/gui/VisualEditorPanel.java [hunk 1]
```diff
@@ -158,14 +158,14 @@ public boolean accept( final File f )
 	 * This, however, does not mean that the underlying {@link InputTriggerConfig} was
 	 * changed at all as the GUI is buffering changes until the "Apply" button is pressed.
 	 */
-	private final Listeners.List< ConfigChangeListener > configChangeListeners;
+	public final Listeners.List< ConfigChangeListener > modelChangedListeners;
```
Comment (tpietzsch) [inline src/main/java/org/scijava/ui/behaviour/io/gui/VisualEditorPanel.java]: Is this by accident? Why should this be public?

File: src/main/java/org/scijava/ui/behaviour/io/gui/VisualEditorPanel.java [hunk 2]
```diff
@@ -158,14 +158,14 @@ public boolean accept( final File f )
...
 
 	/**
 	 * Set of listeners that are triggered only when the "Apply" button is pressed,
 	 * which is precisely the moment when the current state/content of GUI is committed
 	 * to the underlying {@link InputTriggerConfig} via the ModelToConfig().
 	 */
-	private final Listeners.List< ConfigChangeListener > configCommittedListeners;
+	public final Listeners.List< ConfigChangeListener > configCommittedListeners;
```
Comment (tpietzsch) [inline src/main/java/org/scijava/ui/behaviour/io/gui/VisualEditorPanel.java]: Is this by accident? Why should this be public?

**Comment by tpietzsch** [comment]
Please make the fields private again and add back public getters.

Note, that these are (were) not the same type!
The private field is
`Listeners.List<ConfigChangeListener>`
while the getter method returns
`Listeners<ConfigChangeListener>`

`Listeners.List` allows to iterate the registered listeners and send messages to them.
`Listeners` does not allow that.

We expose only `Listeners` via the getter, so that clients can use that to register listeners, but they cannot themselves send events or mess with other listeners.

**Comment by xulman** [comment]
Hi @tpietzsch , thanks for reviewing and esp. for the careful reading... I have totally overlooked last time the "detail", I'm sorry for that.

I've basically reverted and revised again. The current change in this branch is essentially:
- all ill-named `*configChangeListeners*` methods got deprecated,
- and corresponding `*modelChangedListeners*` were added (and referenced & adviced from the deprecated methods)

**Pull request merged**